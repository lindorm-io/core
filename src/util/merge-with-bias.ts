import { TObject } from "../typing";

export const mergeObjectsWithBias = (primary: TObject<any>, secondary: TObject<any>): TObject<any> => {
  const object: TObject<any> = secondary;

  for (const [key, value] of Object.entries(primary)) {
    if (value === null || value === undefined) continue;
    object[key] = value;
  }

  return object;
};
