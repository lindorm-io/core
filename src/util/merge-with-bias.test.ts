import { mergeObjectsWithBias } from "./merge-with-bias";

describe("mergeObjectsWithBias", () => {
  test("should prioritize primary input", () => {
    expect(
      mergeObjectsWithBias(
        { a: "primary", b: null, c: "primary", d: undefined, e: undefined },
        { a: "secondary", b: "secondary", c: null, d: null, e: 12345 },
      ),
    ).toMatchSnapshot();
  });
});
