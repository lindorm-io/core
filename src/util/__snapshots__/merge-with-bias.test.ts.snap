// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mergeObjectsWithBias should prioritize primary input 1`] = `
Object {
  "a": "primary",
  "b": "secondary",
  "c": "primary",
  "d": null,
  "e": 12345,
}
`;
